core/os/input_event.cpp

```diff
@@ -62,10 +62,16 @@ bool InputEvent::is_action(const StringName &p_action) const {
 
 bool InputEvent::is_action_pressed(const StringName &p_action) const {
 
-	return false; // InputMap::get_singleton()->event_is_action(Ref<InputEvent>(this),p_action);
+	if (is_action(p_action)) {
+		return is_pressed();
+	}
+	return false;
 }
 bool InputEvent::is_action_released(const StringName &p_action) const {
 
+	if (is_action(p_action)) {
+		return !is_pressed();
```

Comment:
Well, no, there is only one `pressed=false` event when the key is released. So, this is going to catch just that one